File: longest-subsequence-of-given-difference/longest-subsequence.js
//leetcode - Longest Arithmetic Subsequence of Given Difference - Medium

/**
 * @param {number[]} arr
 * @param {number} difference
 * @return {number}
 */

var longestSubsequence = function(arr, difference) {
    const lookup = new Map();
    for (let i = 0; i < arr.length; i++) {
        if (lookup.has(arr[i] - difference)) {
            lookup.set(arr[i], lookup.get(arr[i] - difference) + 1)
        } else {
            lookup.set(arr[i], 1)
        }
    }
    let longest = 0
    for (const value of lookup.values()) {
        longest = Math.max(longest, value)
    }
    return longest
};

File: longest-subsequence-of-given-difference/longest-subsequence.py
#Longest Arithmetic Subsequence of Given Difference - Medium

import math
class Solution:
    def longestSubsequence(self, arr: List[int], difference: int) -> int:
        lookup = {}
        for num in arr:
            lookup[num] = lookup.get(num - difference, 0) + 1
        longest = 0
        for value in lookup.values():
            longest = max(longest, value)
        return longest